### ISSUE TITLE
Problem with Skill Items

### ISSUE THREAD

**Issue by moepi2k** [suriyun-production/mmorpg-kit-docs#2071]
when using skillitems there should be an option to consume the item only after success cast

https://gyazo.com/cdfbdf41c58871dd73b0545c1a80ad61

else it happen when you use the item and die while casting or use the option caninterruptwhencast the item disappear.

**Comment by insthync** [comment]
More options again?

**Comment by moepi2k** [comment]
I dont know if option or maybe change ornlet it as it is, at the end its ur choice. But right now in some case it can feel like a bug because item disappear without having the effect of the item when us interrupt true

**Comment by insthync** [comment]
Try 1.83d2

**Comment by moepi2k** [comment]
Thanks mate 👍

**Comment by moepi2k** [comment]
ok item decrease after skill casted, but there need to be an addiotonal check to prevent spaming it like isDoingAction
else you can have multible effect of one item while spam it when use cast duration.

**Comment by insthync** [comment]
I can't reproduce it.

**Comment by moepi2k** [comment]
Try use two diffrent skill items with casttime 5 sec
Then use item 1 and while casting use item 2,

In this way u can spam those two items

**Comment by insthync** [comment]
Ahh, I see, 2 difference items

**Comment by moepi2k** [comment]
exactly, i tested it also now with playercharactercontroller and fresh kit. it seems with it you cant reproduce it and let spam two items together. but you can with aimatcursorcontroller.

**Comment by moepi2k** [comment]
that fixxes the issue
https://gyazo.com/6f87126cde475ca8217f4ffeb5dbb80d

**Comment by moepi2k** [comment]
but there is another issue (sorry for that)
but when the cast is interrupted, it seems in the background it still finishing the cast (stuck in place) you have rubberband
and after finish the cast the item disappear without getting the effect

https://gyazo.com/ca84ff5806623884f8a2153f9714ed89

**Comment by insthync** [comment]
Try 1.83e

**Comment by moepi2k** [comment]
Same issue

**Comment by insthync** [comment]
Fixed
